_ko/tour/tour-of-scala.md

```diff
@@ -2,51 +2,66 @@
 layout: tour
 title: 들어가며
 
-discourse: false
+discourse: true
 
 partof: scala-tour
 
 num: 1
-language: ko
 
 next-page: basics
+
+redirect_from: "/tutorials/tour/tour-of-scala.html"
+
+next-page: basics
+
+language: ko
 ---
 
-스칼라는 정확하고 명쾌하며 타입 세이프한 방식으로 일반적인 유형의 프로그래밍 패턴을 표현하기 위해 설계된 새로운 다중 패러다임 프로그래밍 언어다. 스칼라는 객체지향과 함수형 언어를 자연스럽게 통합해준다.
+## 투어를 환영합니다
+이 투어에서는 스칼라에서 가장 자주 사용되는 기능을 요약하여 소개하며, 스칼라 초보자를 그 대상으로 합니다.
+
+완전한 언어 학습서가 아닌 간단한 둘러보기만 제공하므로, 필요하면 [책](/books.html)을 구해보거나 [다른 자료](/learn.html)를 검토하세요.
+
+## 스칼라란?
+스칼라는 일반적인 프로그래밍 패턴을 간결하고 우아하며 타입-세이프한 방식으로 표현할 수 있게 설계된 최신 멀티-패러다임 프로그래밍 언어입니다. 객체지향과 함수형 언어의 특징을 자연스럽게 통합합니다.
 
 ## 스칼라는 객체지향이다 ##
-[모든 값이 객체](unified-types.html)라는 측면에서 스칼라는 순수 객체지향 언어다. 객체의 타입과 행위는 [클래스](classes.html)와 [트레잇](traits.html)으로 나타난다. 클래스는 서브클래스를 만들거나, 다중 상속을 깔끔하게 대체하는 유연한 [믹스인 기반 컴포지션](mixin-class-composition.html) 방법을 통해 확장된다.
+[모든 값이 객체](unified-types.html)라는 의미에서 스칼라는 순수 객체지향 언어입니다. 객체의 타입과 행위는 [클래스](classes.html)와 [트레잇](traits.html)에 의해 설명됩니다. 클래스는 서브클래스를 만들거나, 다중 상속을 깔끔하게 대체하는 유연한 [믹스인 기반 컴포지션](mixin-class-composition.html)을 통해 확장가능합니다.
 
 ## 스칼라는 함수형이다 ##
-또한, 스칼라는 [모든 함수가 값](unified-types.html)이라는 측면에서 함수형 언어다. 스칼라는 익명 함수를 위한 경량 구문을 제공하고, [고차 함수](higher-order-functions.html)를 지원하며, 함수의 [중첩](nested-functions.html)을 허용하고, [커링](multiple-parameter-lists.html)을 지원한다. 스칼라의 [케이스 클래스](case-classes.html)와 케이스 클래스의 [패턴 매칭](pattern-matching.html) 빌트인 지원을 통해 여러 함수형 프로그래밍 언어에서 사용되는 대수 타입을 만들 수 있다.
+또한, 스칼라는 [모든 함수가 값](unified-types.html)이라는 의미에서 함수형 언어입니다. 스칼라는 익명 함수를 위한 경량 구문을 제공하고, [고차 함수](higher-order-functions.html)를 지원하며, 함수의 [중첩](nested-functions.html)을 허용하고, [커링](multiple-parameter-lists.html)을 지원합니다. 스칼라의 [케이스 클래스](case-classes.html)와 케이스 클래스의 [패턴 매칭](pattern-matching.html) 빌트인 지원을 통해 여러 함수형 프로그래밍 언어에서 사용되는 대수 타입을 만들 수 있습니다.
 
-뿐만 아니라 스칼라의 패턴 매칭 개념은 [우측 무시 시퀀스 패턴](regular-expression-patterns.html)의 도움을 받아 자연스럽게 XML 데이터의 처리로 확장된다. 이런 맥락에서 시퀀스 컴프리헨션은 쿼리를 만들 때 유용하다. 이런 기능 때문에 스칼라는 웹 서비스와 같은 애플리케이션 개발에 있어서 이상적인 선택이 될 수 있다.
+뿐만 아니라 스칼라의 패턴 매칭 개념은 [우측 무시 시퀀스 패턴](regular-expression-patterns.html)의 도움을 받아 [extractor 객체](extractor-objects.html)를 이용한 일반적인 확장을 통해 자연스럽게 XML 데이터의 처리로 확장됩니다. 이런 맥락에서 [for 컴프리헨션](for-comprehensions.html)은 쿼리를 만들어 내는데 유용합니다. 이런 기능 덕분에 스칼라는 웹 서비스와 같은 애플리케이션 개발에 있어서 이상적인 선택이 될 수 있습니다.
```

Comment:
I'm not sure if my translation is correct. Let's think together. :)

**뿐만 아니라** 스칼라의 패턴 매칭 개념은 **우측 무시 시퀀스 패턴의 도움을 받아 객체를 이용한 일반적인 확장을 통해 자연스럽게 XML 데이터의 처리로 확장됩니다**.

**또한**, 스칼라의 패턴 매칭 개념은 **추출자 객체를 이용한 일반적인 확장으로 우측 무시 시퀀스 패턴의 도움을 받아 XML 데이터의 처리까지 자연스럽게 확장됩니다**.

Reply:
@byeonggukgong, yours is more natural, I think.
And it should better change '추출자 객체' to '추출자 오브젝트', because it is occurred in other docs.

> I'm not sure if my translation is correct. Let's think together. :)
> 
> **뿐만 아니라** 스칼라의 패턴 매칭 개념은 **우측 무시 시퀀스 패턴의 도움을 받아 객체를 이용한 일반적인 확장을 통해 자연스럽게 XML 데이터의 처리로 확장됩니다**.
> 
> **또한**, 스칼라의 패턴 매칭 개념은 **추출자 객체를 이용한 일반적인 확장으로 우측 무시 시퀀스 패턴의 도움을 받아 XML 데이터의 처리까지 자연스럽게 확장됩니다**.

Reply:
If so, do it.